content/en/docs/Advanced/sys-extensions.md

```diff
@@ -0,0 +1,156 @@
+                        └── k3s.service
+```
+
+Then you can use the `systemd-repart` tool to create the sysext image:
+```bash
+$ systemd-repart -S -s SOURCE_DIR NAME.sysext.raw --private-key=PRIVATE_KEY --certificate=CERTIFICATE       
+```
+
+{{% alert title="Warning" %}}
+Note that the extensions MUST have a `/usr/lib/extension-release.d/extension-release.NAME` file in which the NAME needs to match the sysext NAME (extension is ignored). This is an enforcement by systemd to ensure the sysext is correctly identified and some sanity checks are done with the info in that file.
+{{% /alert %}}
+
+This will generate a signed+verity sysextension that can then be used by sysext to extend the system.
+
+Some extension examples are available under https://github.com/Itxaka/sysext-examples for k3s and sbctl.
+
+
+### Verifying the system extensions
+
+
+You can use `systemd-dissect` to verify the system extension, the ID, ARCHITECTURE and the partitions that are included in the system extension.
+
+```bash
+$ sudo systemd-dissect sbctl-0.14.sysext.raw
+      Name: sbctl-0.14.sysext.raw
+      Size: 21.0M
+ Sec. Size: 512
+     Arch.: x86-64
+
+Image UUID: 351f0e17-35e5-42ff-bf09-8db65c756f7b
+ sysext R.: ID=_any
+            ARCHITECTURE=x86-64
+
+    Use As: ✗ bootable system for UEFI
+            ✗ bootable system for container
+            ✗ portable service
+            ✗ initrd
+            ✓ sysext for system
+            ✓ sysext for portable service
+            ✗ sysext for initrd
+            ✗ confext for system
+            ✗ confext for portable service
+            ✗ confext for initrd
+
+RW DESIGNATOR      PARTITION UUID                       PARTITION LABEL        FSTYPE                AR>
+ro root            4afae1e5-c73c-2f5a-acdc-3655ed91d4e0 root-x86-64            erofs                 x8>
+ro root-verity     abea5f2f-214d-4d9f-83f8-ee69ca7614ba root-x86-64-verity     DM_verity_hash        x8>
+ro root-verity-sig bdb3ee65-ed86-480c-a750-93015254f1a7 root-x86-64-verity-sig verity_hash_signature x8>
+```
+
+
+### Adding system extensions to build medium
+
+Check the `Bundling system extensions during the installable medium build` section in the [Trusted Boot Installations]({{%relref "/docs/Installation/trustedboot#bundling-system-extensions-during-the-installable-medium-build" %}}) for more information on how to add system extensions to the build medium.
+
+
+### Adding system extensions to the installed system
+
+Its possible to add system extensions once the system is installed or as part of the installation process without having to bundle them directly on the install media. This is done by copying the sysexts directly into the EFI partition in the proper place.
+
+The paths to copy the sysexts are:
+- `EFI/kairos/active.efi.extra.d/` for the sysexts that will be loaded when choosing the active boot entry.
+- `EFI/kairos/passive.efi.extra.d/` for the sysexts that will be loaded when choosing the passive boot entry.
+
```

Comment:
maybe worth mentioning that if you want it on both you need to add it on both?

Reply:
Not clear enough? I mean, I think it tries to make it clear that they are by efi file and if you put it on X it will only be loaded by X.

Let me try to rephrase this somehow!

Reply:
hwo about now @mauromorales ?

Reply:
great